From 59eb6694cd464f7a6f0a481919e79c43d98efcdb Mon Sep 17 00:00:00 2001
From: Lianlian Zhu <lianlian.zhu@amlogic.com>
Date: Mon, 4 Nov 2019 09:32:50 +0800
Subject: [PATCH] audio: fixed spdif passthrough failed issue [1/2]

PD#OTT-7128

Problem:
when hdmi connect a unsupport dolby tv and spdif connect
a avr, avr passthough failed.

Solution:
change isPassthroughSupported  api, do  not use hdmi
edid support,use ui supported SurroundFormats.

Verify:
verify by u212

Change-Id: Ia726bf84462f33bc7ff0a2c00b7562a07c6ffa66
Signed-off-by: Lianlian Zhu <lianlian.zhu@amlogic.com>
---
 media/libaudioclient/AudioSystem.cpp | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/media/libaudioclient/AudioSystem.cpp b/media/libaudioclient/AudioSystem.cpp
index 202a82a..9f08e7e 100644
--- a/media/libaudioclient/AudioSystem.cpp
+++ b/media/libaudioclient/AudioSystem.cpp
@@ -1113,7 +1113,15 @@ bool AudioSystem::isPassthroughSupported(const audio_format_t audioFormat)
     } else {
       hdmi_format = 0;
     }
-    if (!property_get_bool("ro.vendor.platform.is.tv", false /* default_value */)) {
+
+    if (hdmi_format == 4) {
+       if (audioFormat == AUDIO_FORMAT_AC3 || audioFormat == AUDIO_FORMAT_DTS)
+           return true;
+       else
+           return false;
+    }
+    if (!property_get_bool("ro.vendor.platform.is.tv", false /* default_value */) 
+        || property_get_bool("ro.vendor.platform.is.stb", false)) {
        hdmi_format = 5;
     }
     if (hdmi_format == 0) {
@@ -1125,7 +1133,7 @@ bool AudioSystem::isPassthroughSupported(const audio_format_t audioFormat)
         } else {
             unsigned int numSurroundFormats;
             bool surroundFormatsEnabled[10];
-            bool reported = true;
+            bool reported = false;
             audio_format_t surroundFormats[10];
             if (NO_ERROR == aps->getSurroundFormats(
                   &numSurroundFormats, surroundFormats, surroundFormatsEnabled, reported)) {
-- 
2.7.4

